pkg/sbom/syft/scanner.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2025 Chainguard, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package syft provides integration with Syft for scanning package contents
+// to enrich SBOMs with detected components.
+package syft
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"chainguard.dev/melange/pkg/sbom"
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/source/directorysource"
+	"github.com/chainguard-dev/clog"
+	purl "github.com/package-url/packageurl-go"
+)
+
+// Scanner wraps Syft functionality for scanning package contents
+type Scanner struct {
+	// Path to scan
+	path string
+}
+
+// NewScanner creates a new Syft scanner for the given path
+func NewScanner(path string) *Scanner {
+	return &Scanner{
+		path: path,
+	}
+}
+
+// Scan performs a Syft scan on the configured path and returns detected packages
+func (s *Scanner) Scan(ctx context.Context) ([]sbom.Package, error) {
+	log := clog.FromContext(ctx)
+	log.Infof("scanning package contents with Syft: %s", s.path)
+
+	// Create a Syft source from the directory
+	// In Melange, we're always scanning directories containing package contents
+	src, err := directorysource.NewFromPath(s.path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Syft source from path %s: %w", s.path, err)
+	}
+	defer src.Close()
+
+	// Configure Syft to scan for all package types with default settings
+	cfg := syft.DefaultCreateSBOMConfig().
+		WithParallelism(4)
```

Comment:
We'll want to configure catalogers so we can find all installed software, which I don't think is the default when using a directory source (above)